tests/logging-utils/test_DEBUG2_logging.py

```diff
@@ -32,3 +33,47 @@ def test_extended_debug_logger(caplog, DEBUG2_enabled):
     assert record.levelno == 8
     assert record.args == (1,)
     assert record.message == "message 1"
+
+
+def test_caching_of_debug2_when_disabled(DEBUG2_installed):
+    # we need a unique logger because loggers are cached
+    logger = logging.getLogger("testing-{}".format(uuid.uuid4()))
+
+    assert logger.isEnabledFor(DEBUG2_LEVEL_NUM) is False
+
+    assert 'show_debug2' not in logger.__dict__
+    assert logger.show_debug2 is False
+    # cached property should have inserted it into the dict
+    assert 'show_debug2' in logger.__dict__
+
+    assert 'debug2' not in logger.__dict__
+    assert logger.debug2('this should actually call the function') is None
+    assert 'debug2' in logger.__dict__
+    assert logger.debug2('should not do anything but hit the lambda') is None
```

Comment:
I can't see anywhere that the test is making sure that the log is being dropped.

Reply:
I'll add a caplog check, forgot I could do that.